modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.txt

```diff
@@ -175,3 +175,13 @@ class org.elasticsearch.index.similarity.ScriptedSimilarity$Doc {
   int getLength()
   float getFreq()
 }
+
+class org.elasticsearch.script.AnalysisScript$Term {
+  CharSequence term
+  int posInc
+  int posLen
+  int startOffset
+  int endOffset
+  String type
+  boolean isKeyword
+}
```

Comment:
I'm exposing this information directly at the moment, but it might be better to expose getters instead so that consumers don't try and do things like change offsets or position increments

Reply:
+1 to exposing getters

Reply:
also should we expoe the absolute position rather than the position increment? This looks more useful to me for filtering. Or both?

Reply:
Painless doesn't require the `get` part of the getter anyway so I'd prefer getters.